src/components/settings/ColorPickerSection.jsx

```diff
@@ -0,0 +1,218 @@
+import React, { useCallback, useEffect, useState } from "react"
+import PropTypes from "prop-types"
+import * as _ from "lodash"
+
+import ColorPicker from "./ColorPicker"
+import FormFieldColor from "./FormFieldColor"
+
+import elementStyles from "../../styles/isomer-cms/Elements.module.scss"
+
+const ColorPickerSection = ({ colors, saveChanges }) => {
+  const [currColors, setCurrColors] = useState(colors)
+  const [colorPicker, setColorPicker] = useState({
+    currentColor: "",
+    elementId: "",
+    oldColors: {},
+  })
+  const [colorPickerToggle, setColorPickerToggle] = useState(false)
+
+  useEffect(() => {
+    setCurrColors(colors)
+  }, [colors])
+  // event listener callback function that resets ColorPicker modal
+  // when escape key is pressed while modal is active
+  const escFunction = useCallback(
+    (event) => {
+      if (event.key === "Escape") {
+        setColorPickerToggle(false)
+      }
+    },
+    [setColorPickerToggle]
+  )
+
+  // event listener callback function that resets ColorPicker modal
+  // when mouse clicks on area outside of modal while modal is active
+  const clickFunction = useCallback(
```

Comment:
Nice feature. Can I make a suggestion that we don't change the color in the background until the user actually presses "Select"? Otherwise it's slightly unintuitive to me that it doesn't "autosave", as in the color in the Settings screen is changing when I change the color in the ColorPicker modal, but when I click outside it doesn't "autosave" and instead reverts back. Maybe this should only change when the user presses "Select" maybe?

https://user-images.githubusercontent.com/39231249/141722461-df5579f7-2ebd-447f-8bfa-cbd3c6298f26.mov

rt

Reply:
@NatMaeTan thoughts?

Reply:
After discussion with @NatMaeTan, we have decided to leave it as is because the colour picker modal does not show the selected colour clearly enough.